### ISSUE TITLE
Can't upload traces with Safari

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#512]
**[Submitted to the original trac issue database at 9.53am, Tuesday, 17th July 2007]**

[http://www.openstreetmap.org/traces/mine Uploading] a GPS trace with Safari fails with:

```
Application error

Change this error message for exceptions thrown outside of an action
(like in Dispatcher setups or broken Ruby code) in public/500.html
```

The error comes back immediately, so I suspect the track just isn't being attached for some reason (!FireFox works as expected, with a short delay from hitting submit to getting the result back).

**Comment by tomhughes** [comment]
**Author: tom@compton.nu**
**[Added to the original trac issue at 10.24am, Tuesday, 17th July 2007]**

Safari for Windows seems to work fine (I don't have access to a Mac to test there).

Can you let me know what IP address you were using, and the rough date/time of the failed upload so I can look for it in the logs?

**Comment by tomhughes** [comment]
**Author: dair@refnum.com**
**[Added to the original trac issue at 6.31am, Wednesday, 18th July 2007]**

I tried to upload "etrex.gpx", description "Cranleigh", at 07:30 in the UK on the 18th (from IP address 212.84.122.95).

**Comment by tomhughes** [comment]
**Author: tom@compton.nu**
**[Added to the original trac issue at 11.47pm, Tuesday, 21st August 2007]**

sorry to be a pain, but I didn't see your previous response before the logs expired - could you try it again and let me have the details. Trac is sending email now so I will actually see your response  straight away this time.

**Comment by tomhughes** [comment]
**Author: tom@compton.nu**
**[Added to the original trac issue at 11.19pm, Wednesday, 22nd August 2007]**

This is being triggered by a bug/misfeature in Safari whereby it sometimes doesn't sent a Content-Type header for an uploaded file. As a result rails winds up trying to treat it as an ordinary string parameter instead of an uploaded file - more details at:

  http://blog.pluron.com/2007/07/safari-30-webki.html
  http://dev.rubyonrails.org/changeset/7005

I need to think about what we can do about this until the rails fix shown above makes it's way into a release.

**Comment by tomhughes** [comment]
**Author: tom@compton.nu**
**[Added to the original trac issue at 1.03pm, Saturday, 23rd February 2008]**

I believe this should have been fixed by the rails 2.0.1 upgrade.
